### ISSUE TITLE
Performance: Maximum update depth exceeded

### ISSUE THREAD

**Issue by absurd-turtle** [ElektraInitiative/PermaplanT#1121]
The performance on the current master is very poor for me.
E.g. when selecting plants and zooming

It seems like some component gets updated extensively.
I get this warning in the console:
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.

**Comment by Bushuo** [comment]
It is the Timeline component, should be fixed with #1119
Mentioned it in #1107

**Comment by markus2330** [comment]
Thank you for reporting the problem and thx @Bushuo for your triage/help.

I fully agree this must be fixed before we do a release, I tagged accordingly.

**Comment by markus2330** [comment]
@absurd-turtle can you check if https://github.com/ElektraInitiative/PermaplanT/pull/1119 fixes the problem?

**Comment by danielsteinkogler** [comment]
I wondered why the whole EditorMap was rerendered when zooming or dragging the map. I debugged it with profiler and i found that EditorMap depends on whole untrackedState althought only some parts were used in the component. so every time the viewRect was changed the whole editorMap and childs have been rendered (also timeline). 
I think i fixed this also here: https://github.com/ElektraInitiative/PermaplanT/pull/1119.
